torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1702,13 +1702,48 @@ TensorExprKernel::BackendType TensorExprKernel::inferBackendTypeFromDevice(
   return backendType;
 }
 
+static bool isValidIdentifierChar(char c, size_t pos) {
+  return islower(c) || isupper(c) || c == '_' || (pos > 0 && isdigit(c));
+}
+
+// replaces all invalid characters with underscore
+std::string sanitizeName(const std::string& input_name) {
+  std::stringstream sanitized_name;
+  for (size_t i = 0; i < input_name.size(); ++i) {
+    if (isValidIdentifierChar(input_name[i], i)) {
+      sanitized_name << input_name[i];
+    } else {
+      sanitized_name << "_";
+    }
+  }
+  return sanitized_name.str();
+}
+
+// we use the debug names in printing cuda code, they need to be removed
+// of characters that can't be used in a variable identifier
+void TensorExprKernel::genInputDebugNames() {
+  std::unordered_map<std::string, const torch::jit::Value*> name_to_value;
+  std::unordered_set<std::string> name_set;
+  std::unordered_map<const torch::jit::Value*, std::string> value_to_name;
+  for (const torch::jit::Value* input : graph_->inputs()) {
+    std::string sanitized_name = sanitizeName(input->debugName());
+    // we could get fancier here, but name conflict is extremely unlikely
+    while (name_set.count(sanitized_name)) {
```

Comment:
We don't really need to make the names unique as TE IR Printer will ensure that (modulo bug #52727 - but that bug doesn't affect correctness, it only affects debuggability). So, we probably could avoid creating the input-name map here.

Reply:
I'm going to comment on that issue to remove this logic as part of the task, i still think for debugability id prefer to keep this in until that lands